packages/asset/contracts/AssetCreate.sol

```diff
@@ -354,5 +374,269 @@ contract AssetCreate is
         return ERC2771HandlerUpgradeable._msgData();
     }
 
-    uint256[45] private __gap;
+    /// @notice Lazily creates a new asset with a signature
+    /// @dev Allows users to lazy mint assets
+    /// @param mintData The data for the lazy mint
+    function lazyCreateAsset(
+        bytes memory signature,
+        LazyMintData calldata mintData,
+        ExchangeMatch[] calldata matchedOrders
+    ) external whenNotPaused {
```

Comment:
you'll need the `from` as first argument to make it work with ApproveAndCall

Reply:
Can you elaborate on this?
Why do I need from as the first argument? I have the caller in the `mintData`

Reply:
You can look at SAND to see how it works,
```
abstract contract ERC20BasicApproveExtension is ERC20Internal, Context {
    /// @notice Approve `target` to spend `amount` and call it with data.
    /// @param target The address to be given rights to transfer and destination of the call.
    /// @param amount The number of tokens allowed.
    /// @param data The bytes for the call.
    /// @return The data of the call.
    function approveAndCall(
        address target,
        uint256 amount,
        bytes calldata data
    ) external payable returns (bytes memory) {
        require(BytesUtil.doFirstParamEqualsAddress(data, _msgSender()), "FIRST_PARAM_NOT_SENDER");

        _approveFor(_msgSender(), target, amount);

        // solhint-disable-next-line avoid-low-level-calls
        (bool success, bytes memory returnData) = target.call{value: msg.value}(data);
        require(success, string(returnData));
        return returnData;
    }
```

Reply:
Oh yeah I see, I will add an extra param, good catch.